Source/CarthageKit/Project.swift

```diff
@@ -1109,6 +1144,24 @@ public final class Project { // swiftlint:disable:this type_body_length
 					}
 			}
 	}
+
+	/// Determines whether the requirements specified in this project's Cartfile.resolved
+	/// are compatible with the versions specified in the Cartfile for each of those projects.
+	///
+	/// Either emits a value to indicate success or an error.
+	public func validate(resolvedCartfile: ResolvedCartfile) -> SignalProducer<(), CarthageError> {
+		return SignalProducer(value: resolvedCartfile)
+			.flatMap(.concat) { (resolved: ResolvedCartfile) -> SignalProducer<([Dependency: PinnedVersion], CompatibilityInfo.Requirements), CarthageError> in
+				return SignalProducer(value: resolved.dependencies).zip(with: self.requirementsByDependency(resolvedCartfile: resolved, tryCheckoutDirectory: true))
```

Comment:
This would be clearer if it were written with the `static func` version of `zip`.

```swift
return SignalProducer.zip(
  SignalProducer(value: resolved.dependencies),
  self.requirementsByDependency(…)
)
```